addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -256,8 +256,10 @@ def _l10n_in_validate_partner(self, partner, is_company=False):
             message.append(_("- Street2 should be min 3 and max 100 characters"))
         if not re.match("^.{3,100}$", partner.city or ""):
             message.append(_("- City required min 3 and max 100 characters"))
-        if partner.country_id.code == "IN" and not re.match("^.{3,50}$", partner.state_id.name or ""):
-            message.append(_("- State required min 3 and max 50 characters"))
+        if partner.country_id.code == "IN" and not re.match("^(?!0+$)([0-9]{1,2})$",
+            partner.state_id.l10n_in_tin or ""
+        ):
```

Comment:
```suggestion
        if partner.country_id.code == "IN" and not re.match("^(?!0+$)([0-9]{2})$",partner.state_id.l10n_in_tin or ""):
```
I see in the code weird things for GSTR-3B where it explicitly checks 04 instead of 4...

Does the state still require a name of 3+ chars?  Maybe we could keep the previous error.

Reply:
> Does the state still require a name of 3+ chars? Maybe we could keep the previous error.
We can but it doesn't make sense because every time we are sending the `l10n_in_tin` to the API request, but for consistency I guess it would be better